### ISSUE TITLE
Add support Tab close and new Tab buttons

### ISSUE THREAD

**Issue by christianvogt** [patternfly/patternfly-react#7108]
**Describe the issue. What is the expected and unexpected behavior?**
OpenShift Console is creating a terminal view with the ability to add and remove tabs. It uses the `Tab` component along with `TabTitleIcon` in the title.

There is no clean way to support the addition of a button in the title. We can easily add our own `Button`, minus the padding. However I feel like the addition of a close button should be part of the react component framework.

Additionally, support for a button to the right of the tabs that is always visible in order to `add new tab` should be provided. Today we can get around this by adding an additional tab and make it function as a button and not a content tab. However if many tabs are present, this tab can scroll off screen. Ideally such a button would become sticky on the right side and not get hidden.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
enhancement

**What is your product and what release version are you targeting?**
OpenShift 4.11

**Comment by mcoker** [comment]
cc @mcarrano @mmenestr

**Comment by mcarrano** [comment]
@christianvogt do you have a mockup of the design you are trying to implement so I could see what this looks like in context?

**Comment by abhinandan13jan** [comment]
![pf-tab-content-issue](https://user-images.githubusercontent.com/24852534/159324822-180e23c6-cc95-4b96-9689-6b02c37f6bba.gif)

**Comment by abhinandan13jan** [comment]
linked issue -> https://github.com/patternfly/patternfly-react/issues/7097

**Comment by mcarrano** [comment]
Thanks for posting this @abhinandan13jan . Now I understand what you are looking for. We will review this and get it on our roadmap. Can you remind me what the code freeze date is for OS 4.11?

**Comment by mcarrano** [comment]
I don't think it will be possible to deliver this by Apr 22, but by June may be possible. Agree that this would be a useful feature.
